presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -131,6 +153,15 @@ fun NavigationGraph(
                 }
             }
 
+            /* FIXME: Issue when landing back from the Edit Screen this will run
+            *         and overwrite the state that was input in the Edit Screen
+            *         and it will fetch from the DB and restore the state from there */
+            it.arguments?.getString(Screen.EventScreen.EVENT_ID)?.let { eventId ->
+                LaunchedEffect(key1 = true) {
+                    eventViewModel.fetchEventById(eventId)
+                }
+            }
```

Comment:
@philipplackner @MKiperszmid I think this is the issue. As will run to get the event from the DB. So this event screen will run when navigating back from the Edit Screen and also when opening from the Agenda Screen. 

What is the best way to handle this case?

Reply:
Why are you using a LaunchedEffect to get the event by id? If the screen gets recomposed, it will get the event again

You can just call it on the viewmodel's init block
That won't get recomposed, so it should work

Reply:
This is what I am doing. The user taps on one of the agenda cards i.e. events
1) Dropdown menu will open and the user selected `Open`
2) I send this event to update the state which contains the agenda item has has been clicked: 
`agendaScreenEvent(AgendaScreenEvent.OnAgendaItemClicked(agendaItem))`
3) In my AgendaViewModel I update the state for the agenda item that has been selected
```
            is AgendaScreenEvent.OnAgendaItemClicked -> {
                _agendaScreenState.update { agendaScreenState ->
                    agendaScreenState.copy(
                        agendaItemClicked = agendaScreenEvent.agendaItem
                    )
                }
            }
```
4) In my AgendaScreen I have this lamba 
`onSelectedEditAgendaItemClicked: (id: String, AgendaType, agendaMenuActionType: AgendaMenuActionType) -> Unit`

4) The result of this is used in the NavigationGraph to send the event ID argument to the EventScreen
```
 AgendaMenuActionType.OPEN -> {
                                navHostController.navigate(route = "${Screen.EventScreen.EVENT_SCREEN}/$id")
                            }

5) I use the LaunchEffect to fetch the event ID from the arguments so this event can be fetched from the DB
```
 it.arguments?.getString(Screen.EventScreen.EVENT_ID)?.let { eventId ->
                LaunchedEffect(key1 = true) {
                    eventViewModel.fetchEventById(eventId)
                }
            }
```

```

Reply:
I'm not saying that logic is bad

I'm not really understanding why you are calling fetchEventById, on the composable, and not directly on the EventViewModel's init block

Reply:
You should pass the id over navArguments, and grab it using savedStateHandle
That's where you can then call the getEventByid using the id from there

Reply:
This is how I did it:



```
 composable(
            route = Route.EDITOR + "/{text_id}/{title}/{body}/{size}",
            arguments = listOf(
                navArgument("body") {
                    type = NavType.StringType
                },
                navArgument("title") {
                    type = NavType.StringType
                },
                navArgument("size") {
                    type = NavType.IntType
                },
                navArgument("text_id") {
                    type = NavType.StringType
                }
            )
        ) {
            val id = it.arguments?.getString("text_id") ?: ""
            EditorScreen(
                title = it.arguments?.getString("title") ?: "Edit Text",
                onBack = {
                    navController.previousBackStackEntry?.savedStateHandle?.set(id, null)
                    navController.popBackStack()
                },
                onSave = { body ->
                    navController.previousBackStackEntry?.savedStateHandle?.set(id, body)
                    navController.popBackStack()
                },
                textSize = (it.arguments?.getInt("size") ?: 26).sp
            )
        }

composable(
            route = Route.TASK + "?date={date}&action={action}&id={id}",
            arguments = listOf(
                navArgument("date") {
                    type = NavType.StringType
                    nullable = true
                    defaultValue = null
                },
                navArgument("action") {
                    type = NavType.StringType
                    nullable = true
                    defaultValue = null
                },
                navArgument("id") {
                    type = NavType.StringType
                    nullable = true
                    defaultValue = null
                }
            ),
            deepLinks = listOf(navDeepLink { uriPattern = DeepLinks.TASK_DETAIL })
        ) {
            val taskTitle = it.savedStateHandle.get<String>("task_title") ?: ""
            val taskDescription = it.savedStateHandle.get<String>("task_description") ?: ""
            DetailTaskScreen(
                taskTitle = taskTitle,
                taskDescription = taskDescription,
                onClose = {
                    navController.navigateUp()
                },
                openEditor = { id, title, body, size ->
                    navController.navigate(Route.EDITOR + "/$id/$title/$body/$size")
                }
            )
        }
```

Reply:
Then on the Task screen I have:
```
 LaunchedEffect(taskTitle, taskDescription) {
        viewModel.onEvent(
            DetailTaskEvent.OnUpdatedInformation(
                taskTitle,
                taskDescription
            )
        )
    }
```

Reply:
And the viewmodel makes the getTaskById when you receive the id over navArguments:
```
  init {
val itemId = savedStateHandle.get<String>("id")
        if (itemId != null) {
...
```